build/build.go

```diff
@@ -62,7 +62,14 @@ func NewBuildContext(installSuffix string, buildTags []string) *build.Context {
 // If an error occurs, Import returns a non-nil error and a nil
 // *PackageData.
 func Import(path string, mode build.ImportMode, installSuffix string, buildTags []string) (*PackageData, error) {
-	return importWithSrcDir(path, "", mode, installSuffix, buildTags)
+	wd, err := os.Getwd()
+	if err != nil {
+		// Getwd may fail if we're in GOARCH=js mode. That's okay, handle
+		// it by falling back to empty working directory. It just means
+		// Import will not be able to resolve relative import paths.
+		wd = ""
```

Comment:
> What's the advantage of having this intermediate state?

There are many.

It keeps the scope of this PR smaller. To resolve #302, that's it.

Having a smaller PR/smaller scope has many benefits and I think it's a good thing to aim for.  A smaller change to review carefully. It means it can be merged sooner. Then, it can be tested by more people sooner, and if there are any regressions, they can be reported sooner. A fix for an unexpected issue can be simple, or it may turn out that there's a big flaw and the best course of action is to revert the entire PR and rethink the approach. That way, the revert only affects #302 and doesn't change any public APIs. It doesn't create ripple effect, forcing reverts in external repos.

It also means the future changes that involve breaking APIs can be more carefully thought out, planned, and implemented, without holding back and being married to the fix to #302. I wanted to take the following next steps (one step at a time):

1. Refactor internal code of `gopherjs/build` package. It's all in a single huge .go file, and the order of identifiers doesn't follow a very logical pattern. I find it's hard to read/understand. To implement this fix, I first performed this change to better understand what was going on. Then, I applied and tested the fix. Then, once I understood it, I backported it to `master` and made this pretty small PR with an easy to read diff. But before I do anything else, I want to follow up by applying this refactor next.

2. Next, I want to mark the `build`/`compiler`/etc. APIs as not subject to compatibility promise and may change. Similar to how https://godoc.org/golang.org/x/build/gerrit API is documented:

    > The API is not subject to the Go 1 compatibility promise and may change at any time.

    This is unfortunate, but feels neccessary to make progress. The current APIs were unofficially always meant to be private and able to change, but not explicitly documented that way. This would be about making that more clear, and then being able to take advantage of it to improve the code and APIs.

3. Next, I want to move https://godoc.org/github.com/shurcooL/gopherjslib and https://godoc.org/github.com/shurcooL/go/gopherjs_http into gopherjs repo (also without promise of compatibility and ability to change at any time). The goal is to de-duplicate and clean up that functionality once it's all in one repo, where atomic operations can be done, and everything can be tested via Circle CI. (This was something I asked if you'd be okay in the past, and you told me you were fine with it at that time.)

4. Then, once it's in the same repo, I want to refactor/clean up the `build`/`compiler`/`gopherjslib`/`gopherjs_http` APIs, which would certainly include having the working directory be passed to `Import` (just like `go/build.Import`) instead of `build` package trying to figure it out on its own. I want to make it possible for `gopherjs serve` to reuse the same functionality from `gopherjs_http` instead of an unexported implementation within `tool.go`.

That's the general direction I wanted to move, and it's not something I can realistically make happen as part of 1 PR.

If there are sudden pressing needs (for functionality reasons), we can take a shortcut and make API changes sooner, but that should be done in context of a specific requirement.

In summary, if it's possible to fix #302 in an **acceptable way** (i.e., no problems with functionality) without changing public APIs, I want this PR not to change public APIs. If there are functionality problems (i.e., some sort of problem with behavior) that require public APIs changed in order to fix #302, then I have absolutely no problem with doing that, since it's neccessary. But I don't want to increase the scope of this PR to include improvements to the API without a good reason - I don't think that's neccessary nor a good idea.

Does that sound reasonable, or do you think a different approach is better (if so, why)?